ServiceTests/ProductServiceTest.cs

```diff
@@ -0,0 +1,108 @@
+using CustomerAleksandr.TestgRPCApplication.Services;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using Grpc.Core.Testing;
+using Grpc.Core.Utils;
+using Logic.Interfaces;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ServiceTests
+{
+    public class ProductServiceTest
+    {
+        private Mock<IProductService> _mockProductService;
+        private ProductService _productService;
+        private ServerCallContext _testServerCallContext;
+
+        [SetUp]
+        public void Setup()
+        {
+            _mockProductService = new Mock<IProductService>();
+            _mockProductService.Setup(m => m.AddProduct(It.IsAny<Logic.Entities.Product>())).Returns(1);
+            _mockProductService.Setup(m => m.GetAll()).Returns(new List<Logic.Entities.Product> { new Logic.Entities.Product { Title = "Title", Count = 10, Price = 12, Id = 1 } });
+            _mockProductService.Setup(m => m.DeleteProduct(1));
+            _mockProductService.Setup(m => m.BuyProduct(It.IsAny<int>(), It.IsAny<int>()));
+            _mockProductService.Setup(m => m.GetProductById(1)).Returns(new Logic.Entities.Product { Title = "Title", Count = 10, Price = 12, Id = 1 });
+
+            _testServerCallContext = TestServerCallContext.Create("testMethod", null, DateTime.UtcNow.AddHours(1), new Metadata(), CancellationToken.None, "127.0.0.1", null, null, (metadata) => TaskUtils.CompletedTask, () => new WriteOptions(), (writeOptions) => { });
```

Comment:
Create context for each service call and use method name in `TestServerCallContext.Create`